### PR TITLE
(#385) CMS Translation of CTHP tests 

### LINKED ISSUE #385: CMS: Translation of Cancer Type Home Page
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: [title]
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #385 

Added test scenarios to CancerTypeHomePage.feature.

### CODE REVIEW

**Comment by alinai** [comment]
npm test run on dev-ac:
image

cypress test run on dev-ac:
image

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/CMS/$CancerTypeHomePage/index.js [hunk 1]
```diff
@@ -286,4 +286,42 @@ Then('the promo image is matching the earlier selected image', () => {
...
+    cy.get(`input[value='${addCardSection}']`).should('be.visible');
+});
+
+Given('user is navigating to the front end site with spanish path {string} site section plus {string}', (spPath, purl) => {
+    cy.visit(`${frontEndBaseUrl}${spPath}${siteSection}/${purl}-${randomNum}`, { retryOnStatusCodeFailure: true });
+});
+
+And('the following cards have multiple spanish links that start with {string}', (startLink, dataTable) => {
+    for (let { cardType } of dataTable.hashes()) {
+        cy.get(`div[class*="${cardType}"]`).find('a').should('have.length.above', 1);
```
[the remark this answers is not in the document]
Comment (alinai) [reply cypress/e2e/CMS/$CancerTypeHomePage/index.js]: done

File: cypress/e2e/CMS/$CancerTypeHomePage/index.js [hunk 2]
```diff
@@ -286,4 +286,42 @@ Then('the promo image is matching the earlier selected image', () => {
...
+        cy.get(`select[id*="edit-field-audience"] option:contains("${option}")`).should('be.visible');
+    }
+});
+
+And('{string} is present', (audienceToggle) => {
+    cy.get(`summary.seven-details__summary span:contains("${audienceToggle}")`).should('be.visible');
+});
+
+And('the following cards fields are displayed with remove button translated as {string}', (removeBtn, dataTable) => {
+    cy.get(`input[value="${removeBtn}"]`).should('be.visible');
```
[the remark this answers is not in the document]
Comment (alinai) [reply cypress/e2e/CMS/$CancerTypeHomePage/index.js]: @KateMashkinaNIH Updated this with the following but still getting error
And('the following cards fields are displayed with remove button translated as {string}', (removeBtn, dataTable) => {
    for (let { card } of dataTable.hashes()) {
        cy.get(`div[id*='top-paragraph-type-title']:contains("${card}")`).each($el => {
            cy.wrap($el).parent().find(`input:contains("${removeBtn}")`).should('be.visible');
        })
    }
});

image

**Comment by alinai** [comment]
Cypress test run on test-ac:
image

**Comment by alinai** [comment]
Added the following step to $PressRelease -> index.js

And('user clicks on title with url spanish path {string} site section plus {string}', (spPath, purl) => {
    cy.get(`a[href='${spPath}${siteSection}/${purl}-${randomNum}']`).click();
});

cypress test run on test-ac:
image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**